File: Python/1963-minimum-number-of-swaps-to-make-the-string-balanced.py
class Solution:
    def minSwaps(self, s: str) -> int:
        output, count = 0, 0
        for ch in s :
            if ch == '[' :
                count += 1
            else :
                if count <= 0 :
                    count += 1
                    output += 1
                else :
                    count -= 1
        return output

File: Java/1963-minimum-number-of-swaps-to-make-the-string-balanced.java
class Solution {
    public int minSwaps(String s) {
        int open = 0, swaps = 0;
        for (var ch : s.toCharArray()) {
            if (ch == '[') {
                open++;
            }
            else {
                if (open <= 0) {
                    swaps++;
                    open++;
                }
                else {
                    open--;
                }
            }
        }
        return swaps;
    }
}